### ISSUE TITLE
Update with join mapping error: 'Table '[dbo].[TestRow]' not found.'

### ISSUE THREAD

**Issue by igortech** [linq2db/linq2db#1431]
Describe your issue.

If you have a question, first try to check our [wiki](https://github.com/linq2db/linq2db/wiki) and search in [issues](https://github.com/linq2db/linq2db/issues) - maybe your question already answered there.

If you are seeing an exception, include the full exceptions details (message and stack trace).

```
Exception message: LinqToDB.SqlQuery.SqlException: 'Table '[dbo].[TestRow]' not found.'
Stack trace:
   at LinqToDB.SqlProvider.BasicSqlBuilder.BuildExpression(ISqlExpression expr, Boolean buildTableName, Boolean checkParentheses, String alias, Boolean& addAlias, Boolean throwExceptionIfTableNotFound) in C:\projects\linq2db\Source\LinqToDB\SqlProvider\BasicSqlBuilder.cs:line 2136
   at LinqToDB.SqlProvider.BasicSqlBuilder.BuildColumnExpression(SelectQuery selectQuery, ISqlExpression expr, String alias, Boolean& addAlias) in C:\projects\linq2db\Source\LinqToDB\SqlProvider\BasicSqlBuilder.cs:line 461
   at LinqToDB.SqlProvider.BasicSqlBuilder.BuildUpdateSet(SelectQuery selectQuery, SqlUpdateClause updateClause) in C:\projects\linq2db\Source\LinqToDB\SqlProvider\BasicSqlBuilder.cs:line 527
   at LinqToDB.SqlProvider.BasicSqlBuilder.BuildUpdateQuery(SqlStatement statement, SelectQuery selectQuery, SqlUpdateClause updateClause) in C:\projects\linq2db\Source\LinqToDB\SqlProvider\BasicSqlBuilder.cs:line 205
   at LinqToDB.SqlProvider.BasicSqlBuilder.BuildSql(Int32 commandNumber, SqlStatement statement, StringBuilder sb, Int32 indent, Boolean skipAlias) in C:\projects\linq2db\Source\LinqToDB\SqlProvider\BasicSqlBuilder.cs:line 81
   at LinqToDB.SqlProvider.BasicSqlBuilder.BuildSql(Int32 commandNumber, SqlStatement statement, StringBuilder sb, Int32 startIndent) in C:\projects\linq2db\Source\LinqToDB\SqlProvider\BasicSqlBuilder.cs:line 68
   at LinqToDB.Data.DataConnection.QueryRunner.GetCommand(DataConnection dataConnection, IQueryContext query, Int32 startIndent) in C:\projects\linq2db\Source\LinqToDB\Data\DataConnection.QueryRunner.cs:line 182
   at LinqToDB.Data.DataConnection.QueryRunner.SetQuery() in C:\projects\linq2db\Source\LinqToDB\Data\DataConnection.QueryRunner.cs:line 266
   at LinqToDB.Linq.QueryRunnerBase.SetCommand(Boolean clearQueryHints) in C:\projects\linq2db\Source\LinqToDB\Linq\QueryRunnerBase.cs:line 68
   at LinqToDB.Data.DataConnection.QueryRunner.ExecuteNonQuery() in C:\projects\linq2db\Source\LinqToDB\Data\DataConnection.QueryRunner.cs:line 330
   at LinqToDB.Linq.QueryRunner.NonQueryQuery(Query query, IDataContext dataContext, Expression expr, Object[] parameters) in C:\projects\linq2db\Source\LinqToDB\Linq\QueryRunner.cs:line 750
   at LinqToDB.Linq.QueryRunner.<>c__DisplayClass28_0.<SetNonQueryQuery>b__0(IDataContext db, Expression expr, Object[] ps) in C:\projects\linq2db\Source\LinqToDB\Linq\QueryRunner.cs:line 743
   at LinqToDB.Linq.ExpressionQuery`1.System.Linq.IQueryProvider.Execute[TResult](Expression expression) in C:\projects\linq2db\Source\LinqToDB\Linq\ExpressionQuery.cs:line 165
   at MentolLogic.Test.Data.DataModelTest.DataModelButchUpdateTest1() in H:\Projects\RusMonitoring\MentolCiscoAXL\MentolLogic\MentolLogic.Test\Data\DataModelTest.cs:line 142
```
### Steps to reproduce
Include a complete code listing (or project/solution) that we can run to reproduce the issue.

Partial code listings, or multiple fragments of code, will slow down our response or cause us to push the issue back to you to provide code to reproduce the issue.

```c#
        [TestMethod]
        public void DataModelButchUpdateTest1()
        {            
            using (var dm = new DataConnection("PostgreSQL.9.5", "Server=192.168.80.101;Port=5434;Database=mentol18;User Id=user;Password=password"))
            {
                dm.DropTable<TestRow>(throwExceptionIfNotExists: false);
                dm.CreateTable<TestRow>();
                using (var temp = dm.CreateTempTable<TestRow>("tmpTable"))
                {
                    var join = dm.GetTable<TestRow>().InnerJoin(temp, (outer, inner) => outer.Id == inner.Id,
                    (outer, inner) => new
                    {
                        Outer = outer,
                        Inner = inner
                    });

                    var updatable = join.AsUpdatable();
                    updatable = updatable.Set(pair => pair.Outer.Name, pair => pair.Inner.Name);
                    updatable = updatable.Set(pair => pair.Outer.Description, pair => pair.Inner.Description);
                    updatable.Update();
                }
          }
   }

    [Table("TestRow", Schema = "dbo")]
    class TestRow
    {
        [PrimaryKey]
        public int Id;

        [Column]
        public string Name;

        [Column]
        public string Description;
    }

```

### Environment details
linq2db version: *2.5.4*
Database Server: *PostgreSQL 9.6.3*
Database Provider: *PostgreSQL 9.5*
Operating system: *Windows 8.1*
Framework version: *.NET Core 2.1*

**Comment by igortech** [comment]
this error only PostgreSQL provider

**Comment by sdanyliv** [comment]
@igortech, i has been fixed a long time ago but not yet merged https://github.com/linq2db/linq2db/pull/1278.

Could you please check your sources against this branch?

**Comment by igortech** [comment]
yes, branch dev-update_from_postgresql its working

**Comment by MaceWindu** [comment]
released in 3.0.0-preview.1
